integration-tests/groovy-dsl/pom.xml

```diff
@@ -100,6 +120,36 @@
                 </dependency>
             </dependencies>
         </profile>
+        <profile>
+            <id>native</id>
+            <activation>
+                <property>
+                    <name>native</name>
+                </property>
+            </activation>
+            <properties>
+                <quarkus.package.type>native</quarkus.package.type>
+                <!-- To prevent getting DeletedElementException: Unsupported method java.lang.Class.registerNatives()
+                is reachable for some use cases -->
+                <quarkus.native.additional-build-args>--report-unsupported-elements-at-runtime</quarkus.native.additional-build-args>
```

Comment:
Is this suppressing potential issues that users may see?

Reply:
I think we have to handle this carefully. So users need to have `--report-unsupported-elements-at-runtime` in the applications in native build? Is there any chance to remove it in the future?

Reply:
Yeah, I know, fortunately, this flag is only needed for some use cases not all. I may be wrong but I'm afraid that it can only be fixed at Graal VM level, indeed the method `Class#registerNatives` is private/native and it is called only on static initialization of the class `Class` and I don't know any way to prevent that, IMHO it should never occur.

Reply:
If it is really a showstopper for you, the best I can do is to remove the use cases that need this flag and add a note in the limitations or simply keep it like this and add a note in the doc indicating in which cases it could be needed. WDYT?

Reply:
Please also note that since it should never occur, no exception will ever be thrown at runtime.

Reply:
It should be better to remove this flag and add a note in the limitations for such use cases which need it. Also if

- https://github.com/quarkusio/quarkus/issues/2720 get resolved and can we have a chance to leverage on it?

Reply:
I've just pushed some changes that remove the flag and add a note in the limitations

Reply:
Supporting Groovy directly in Quarkus won't be easy, I believe that they will need to limit the support to Groovy classes that can be statically compiled which is a big limitation. Here, as we are already limited to route definitions, I did not have to add this limitation.

Reply:
Thanks @essobedo and it looks good to me.